jerry-core/ecma/base/ecma-helpers-conversion.c

```diff
@@ -620,6 +670,30 @@ ecma_utf8_string_to_number (const lit_utf8_byte_t *str_p, /**< utf-8 string */
     return sign ? -ECMA_NUMBER_ZERO : ECMA_NUMBER_ZERO;
   }
 
+  /** fraction_uint64 stores the convertable number in raw format without decimal dot so
+    * we should divive it with the appropriate power of 10. This division does not cause significant
+    * inaccuracy due to same order of magnitude of the divisible and the divisor number. */
+
+  /** If the convertable number is bigger than Number.MAX_VALUE
+    * then return with +- Infinity */
+  if (!e_in_lit_sign &&
+      (e_in_lit > NUMBER_MAX_VALUE_EXPONENT ||
+      (e_in_lit == NUMBER_MAX_VALUE_EXPONENT &&
+       (double) fraction_uint64 / pow (10, digits - (dot_position == 0 ? 1 : dot_position)) > NUMBER_MAX_VALUE_BASE)))
```

Comment:
I'm not completely convinced that this can be  acomplished using FP arithmetics only. Please try to stick with integers.

Reply:
It may seem that I'm having conversations with myself, but: I have been pondering on this issue further and something(s) do(es) not seem right.
- I'm not completely sure that `dot_position` is needed. `e` should encode exactly how `fraction_uint64` should be interpreted (where to move the decimal point).
- I'm not sure that `e_in_lit` and `e_in_lit_sign` are OK to be used. The 'lit' variants contain the values as they are in the literal. However, that means that multiple 'lit' values can encode the same numeric value. E.g.,
  - `2e308` will become `fraction_uint64 = 2` `e_in_lit = 308` `e = 308`, but 
  - `2.0e308` will be `fraction_uint64 = 20` `e_in_lit = 308` `e = 307`, and
  - `0.2e309` will be `fraction_uint64 = 2` `e_in_lit = 309` `e = 308`, and
  - `20e307` will be `fraction_uint64 = 20` `e_in_lit = 307` `e = 307`.
(If I got everything right.) I'm not sure that the second and fourth example literals will be turned to INF here (but might be because of the dot_position and pow part). However, I think there is a bigger issue:
  - `1e308` will be `fraction_uint64 = 1` `e_in_lit = 308` `e = 308`, but
  - `0.1e309` will be `fraction_uint64 = 1` `e_in_lit = 309` `e = 308`.
And this will turn the second example to INF (because `e_in_lit >308`)!
- As the original issue was judged as not a bug but a performance issue "only", I think that we do not have to be completely precise. That is, we know that the code is correct just slow for some big `e`s, so we don't necessarily have to deal with "not that big" `e`s here. I.e., if the exponent is certainly above 308 (say: 309), then we return early with infinity, but if the exponent is 308 then we let the rest of the code deal with the subtleties whether it will be INF or not (i.e., whether the mantissa is above 1.79... or not). (This comment is modulo the above, meaning that we should be careful in deciding how to interpret "exponent".)

Reply:
Your facts are right. I've got an idea how to accomplish it without using FP arithmetics. I'm going to refactor this part.